cli/cmd/plugin/unmanaged-cluster/tanzu/tanzu.go

```diff
@@ -26,21 +26,25 @@ import (
 	logger "github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster/log"
 	"github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster/packages"
 	"github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster/tkr"
+
+	"github.com/vmware-tanzu/community-edition/cli/cmd/plugin"
 )
 
 //nolint
 const (
-	configFileName        = "config.yaml"
-	bootstrapLogName      = "bootstrap.log"
-	bomDir                = "bom"
-	tkgSysNamespace       = "tkg-system"
-	tkgSvcAcctName        = "core-pkgs"
-	tkgCoreRepoName       = "tkg-core-repository"
-	tkgGlobalPkgNamespace = "tanzu-package-repo-global"
-	tceRepoName           = "community-repository"
-	tceRepoURL            = "projects.registry.vmware.com/tce/main:v0.11.0"
-	outputIndent          = 3
-	maxProgressLength     = 4
+	configFileName           = "config.yaml"
+	bootstrapLogName         = "bootstrap.log"
+	bomDir                   = "bom"
+	compatibilityDir         = "compatibility"
+	tkgSysNamespace          = "tkg-system"
+	tkgSvcAcctName           = "core-pkgs"
+	tkgCoreRepoName          = "tkg-core-repository"
+	tkgGlobalPkgNamespace    = "tanzu-package-repo-global"
+	tceCompatibilityRegistry = "projects.registry.vmware.com/tce/compatibility"
+	tceRepoName              = "community-repository"
+	tceRepoURL               = "projects.registry.vmware.com/tce/main:v0.11.0"
```

Comment:
As part of this work, I'd like to see the user-managed package repo be put in the TKr going forward.

For example, here's an updated version of our TKr.

```yaml
  tkg-core-packages:
  - version: v1.21.2+vmware.1-tkg.1
    images:
      tanzuCorePackageRepositoryImage:
        imagePath: repo-10
        tag: 0.10.0
        repository: projects.registry.vmware.com/tce
      tanzuUserPackageRepositoryImage:
        imagePath: main
        tag: 0.11.0
        repository: projects.registry.vmware.com/tce
```

Reply:
To test this out, feel free to work off a new TKr you create like `project.registry.vmware.com/tce/tkr/v0.17.0-dev+1`

Reply:
Hmmmm I think that the `tceRepoUrl` const in the `tanzu` package was some leftover from work done to make `additionalPackageRepos` a configurable option via the config.

So how should that play in here? A few thoughts:
- The config by default is empty and it pulls from the TKr instead
- What would the behavior be if the user provides their own `additionalPackageRepos` in a config file or via flag?
  - My guess would be it ignores the user managed package repo in the TKr and uses the ones provided by the user

https://github.com/vmware-tanzu/community-edition/blob/7132015da9057fa45d0c36178d7651478de0599d/cli/cmd/plugin/unmanaged-cluster/config/config.go#L54-L56

Reply:
good points. let's talk this one through synchronously today.